### PR TITLE
remove void* m_userdata

### PR DESCRIPTION
**Pull request by pavel-pimenov**

### CODE REVIEW

File: include/libtorrent/disk_io_thread.hpp [hunk 1]
```diff
@@ -516,10 +516,6 @@ namespace aux {
 
 		aux::session_settings m_settings;
 
-		// userdata pointer for the complete_job function, which
-		// is posted to the network thread when jobs complete
-		void* m_userdata;
```
Comment (arvidn) [inline include/libtorrent/disk_io_thread.hpp]: how is the context for the complete_job function passed along now?

**Comment by arvidn** [comment]
?

**Comment by arvidn** [comment]
why did you close this?

**Comment by pavel-pimenov** [comment]
i thought m_userdata is used somewhere.

#ifndef TORRENT_DISABLE_UDATA_COMPLETE_JOB
 void* m_userdata;
#endif
?

**Comment by arvidn** [comment]
implied in my question was: is it a mistake that this member is not used? (probably not). Are there any other things that should be removed too? like other occurrences of that macro perhaps?

**Comment by pavel-pimenov** [comment]
hm. i found m_userdata by accident :( maybe my mistake
where to find an example of use complete_job and m_userdata?

**Pull request merged**